20TH-JP/c100236019.lua

```diff
@@ -0,0 +1,89 @@
+--Number 39: Utopia Double
+--Logical Nonsense
+
+--Substitute ID
+local s,id=GetID()
+
+function s.initial_effect(c)
+	--Must be properly summoned
+	c:EnableReviveLimit()
+	--Xyz summon method
+	aux.AddXyzProcedure(c,nil,4,2)
+	--Search and Xyz
+	local e1=Effect.CreateEffect(c)
+	e1:SetDescription(aux.Stringid(id,0))
+	e1:SetCategory(CATEGORY_TOHAND+CATEGORY_SEARCH+CATEGORY_SPECIAL_SUMMON)
+	e1:SetType(EFFECT_TYPE_QUICK_O)
+	e1:SetCode(EVENT_FREE_CHAIN)
+	e1:SetHintTiming(0,TIMING_END_PHASE)
+	e1:SetRange(LOCATION_MZONE)
+	e1:SetCountLimit(1+id)
+	e1:SetCost(s.cost)
+	e1:SetTarget(s.target)
+	e1:SetOperation(s.operation)
+	c:RegisterEffect(e1)
+end
+	--Detach 1 material
+function s.cost(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return e:GetHandler():CheckRemoveOverlayCard(tp,1,REASON_COST) end
+	e:GetHandler():RemoveOverlayCard(tp,1,1,REASON_COST)
+end
+	--Look for "Double or Nothing!"
+function s.thfilter(c)
+	return c:IsCode(94770493) and c:IsAbleToHand()
+end
+	--Check for "Utopia" Xyz monster, excluding "Number 39: Utopia Double"
+function s.spfilter(c,e,tp,mc)
+	return c:IsSetCard(0x107f) and not c:IsCode(id) and mc:IsCanBeXyzMaterial(c)
+		and c:IsCanBeSpecialSummoned(e,SUMMON_TYPE_XYZ,tp,false,false)
+end
+	--Activation legality
+function s.target(e,tp,eg,ep,ev,re,r,rp,chk)
```

Comment:
In effects with the 'then' conjunction, A is never required for B, so if you resolve A and cannot resolve B then doesn't the effect stop there? So I'm not sure if there needs to be a condition for B here